### DISCUSSION TITLE
Change VSCode extension versions to match the CLI?

### DISCUSSION THREAD

**Discussion by lgarron** [rome/tools#3606] [Suggestions]
While encountering and reporting issues like https://github.com/rome/tools/issues/3603 and https://github.com/rome/tools/issues/3507 , it can be a bit of work to tell if your current versions of the VSCode extension & CLI are expected to match.

Now that the CLI has stable version numbers, would there be any issues with updating the extension to be in sync, and to be released with a version number matching its included LSP?

**Comment by MichaReiser** [comment]
That's a great point! The extension uses the Rome version installed in your project. However, there's currently a known limitation: the extension doesn't restart the LSP after you perform an update. That's something that we have to fix. 

Using the same version numbers won't be possible because we plan to ship preview builds in the future (ideally, nightlies) and the VS marketplace doesn't support preview tags. 

Would it help if the extension shows the Rome server version number in the status bar?

**Reply by lgarron** [reply]
> The extension uses the Rome version installed in your project.

Are you certain? The `v0.18.0` extension disagrees with the CLI on `v10.0.0` as well as on `v0.10.1-next.e8dd9b4`, even after a full reload of VSCode for either. In fact, it seems impossible for me right now to get the latest VSCode extension to avoid showing errors that the CLI is fine with.

> Would it help if the extension shows the Rome server version number in the status bar?

I think that would certainly help!

**Reply by lgarron** [reply]
> In fact, it seems impossible for me right now to get the latest VSCode extension to avoid showing errors that the CLI is fine with.

In fact, things seem to have gotten worse. Due to https://github.com/rome/tools/issues/3507 I need to completely disable the extension to avoid bad autoformatting across my entire project. :-/

I think one issue is that the LSP seems to persist in unexpected ways, across restarts and attempted `npx rome stop` attempts. This is really tricky to debug right now, so I'm actually going to revert my extension to v0.16.1 to avoid further undesirable effects.

**Reply by MichaReiser** [reply]
> Are you certain?

You're right. It does not. :open_mouth:  There has been some misunderstanding. I'll update the documentation and create an issue (#3635). I'll further create an issue to automatically stop the LSP when the last client disconnects (at least, when the LSP was started by the extension) #3636
